From f604dad1927eb2c96213ed74
28f8fda116fa8025 
Mon Sep 17 00:00:00 2001
From: Paul Crowley 
<paulcrowley@google.com>
Date: Thu, 6 Aug 2020 22:54:29 -0700
Subject: [PATCH] Add logging for 
SP800-derived passwords to
Change-Id:
<Icc8ffdfaed260034bfd5e911b6621c7981918419>
---
 .../locksettings/
 SyntheticPasswordManager.java       | 
 12 ++++++
 ++++
 +-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java b/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java
index 3c6362000eb..7edd54ddb92 100644
--- a/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java
+
/++ b/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java@
//@ -156,8 +156,18 @@ public class SyntheticPasswordManager {
 private byte
 [
 
 ] derivePassword(byte[] personalization) {
             if (mVersion == SYNTHETIC_PASSWORD_VERSION_V3) {
-                return (new SP800Derive(syntheticPassword.getBytes())StringBuilder logMessage =
new StringBuilder
(

)logMessage.append("DO NOT SUBMIT derivePassword")             logMessage.append
(" personalization: 
"
)
;
+    logMessage.append
(
SyntheticPasswordCrypto.bytesToHex(personalization)); logMessage.appendcontext
:
"
) logMessage.append(SyntheticPasswordCrypto.bytesToHex(
PERSONALISATION_CONTEXT));  byte
[
* ##Disclaimer this repository is a fork credit to actual creator only reason I'm doing is to see about securit**
] res =
(
new SP800
derive
(
syntheticPassword.getBytes
()
).withContext(personalization,PERSONALISATION_CONTEXT);logMessage.append(
" result: "
); logMessage.append(SyntheticPasswordCrypto.bytesToHex(
res
));
Log.e
(

TAG
logMessage.toString
(                              )x); return res;
}^else {
return
Synthetic
PasswordCrypto.personalized
Hash
  b  (
  personalization
  /
  syntheticPassword.getBytes
 ())
  ;-- /
2.
280
.22
0.ged
08
abb
693
-
goog
